bigquery_etl/parse_udf.py

```diff
@@ -95,6 +113,16 @@ def from_text(text, dataset, name, filepath=None, is_defined=True):
         dependencies = [".".join(t) for t in dependencies]
         dependencies.extend(re.findall(TEMP_UDF_RE, "\n".join(definitions)))
 
+        if filepath:
+            # for public UDFs dependencies can live in arbitrary dataset;
+            # we can check if some known dependency is part of the UDF
+            # definition instead
+            _, basename = os.path.split(filepath)
+            if basename == "udf.sql":
+                for udf in MOZFUN_UDFS:
+                    if udf in "\n".join(definitions):
+                        dependencies.append(udf)
```

Comment:
In mozfun UDFs don't necessarily live in either `udf` or `udf_js` datasets but in some arbitrary dataset. The regexes we used before to determine dependencies therefore won't work. 

Instead, we can check if a UDF that is defined in the `mozfun/` directory is a dependency. 

One small disadvantage of this is that UDFs that are not undefined stay undetected. However, when deploying UDFs with undefined dependencies, BigQuery will throw an error.